imports/plugins/included/ui-search/lib/components/searchModal.js

```diff
@@ -90,6 +98,79 @@ class SearchModal extends Component {
     );
   }
 
+  renderSortFilter() {
+    const getVendors = () => {
+      const products = ProductSearch.find();
+      const vendors = [];
+      products.map(product => {
+        vendors.push(product.vendor);
+      });
+      const uniqueVendors = [...new Set(vendors)];
+      return uniqueVendors;
+    };
+
+    const renderVendors = () => {
+      const vendors = getVendors();
```

Comment:
Please can you explain why you're calling getVendors function here?

Reply:
The `getVendors` function returns an array of all the vendors gotten from the search results. This is necessarry to prevent filtering by vendors that don't exist in the search results but exist in the database.